subsys/bluetooth/controller/ll_sw/ull_adv_sync.c

```diff
@@ -231,25 +263,176 @@ uint8_t ll_adv_sync_ad_data_set(uint8_t handle, uint8_t op, uint8_t len,
 	}
 #endif /* CONFIG_BT_CTLR_DF_ADV_CTE_TX */
 
+	/* Prepare the AD data as parameter to update in PDU */
 	/* Use length = 0 and NULL pointer to retain old data in the PDU.
 	 * Use length = 0 and valid pointer of `data` (auto/local variable) to
 	 * remove old data.
 	 * User length > 0 and valid pointer of `data` (auto/local variable) to
 	 * set new data.
 	 */
-	if (op == BT_HCI_LE_EXT_ADV_OP_UNCHANGED_DATA) {
-		hdr_data[ULL_ADV_HDR_DATA_LEN_OFFSET] = 0U;
-		(void)memset((void *)&hdr_data[ULL_ADV_HDR_DATA_DATA_PTR_OFFSET],
-			     0U, ULL_ADV_HDR_DATA_DATA_PTR_SIZE);
+	val_ptr = hdr_data;
+	if (op == BT_HCI_LE_EXT_ADV_OP_INTERM_FRAG ||
+	    op == BT_HCI_LE_EXT_ADV_OP_LAST_FRAG ||
+	    op == BT_HCI_LE_EXT_ADV_OP_UNCHANGED_DATA) {
+		*val_ptr++ = 0U;
+		(void)memset((void *)val_ptr, 0U,
+			     ULL_ADV_HDR_DATA_DATA_PTR_SIZE);
 	} else {
-		hdr_data[ULL_ADV_HDR_DATA_LEN_OFFSET] = len;
-		(void)memcpy((void *)&hdr_data[ULL_ADV_HDR_DATA_DATA_PTR_OFFSET],
-			     &data, ULL_ADV_HDR_DATA_DATA_PTR_SIZE);
+		*val_ptr++ = len;
+		(void)memcpy(val_ptr, &data, sizeof(data));
 	}
 
-	err = ull_adv_sync_pdu_set_clear(lll_sync, pdu_prev, pdu,
-					 ULL_ADV_PDU_HDR_FIELD_AD_DATA, 0,
-					 hdr_data);
+	if (!IS_ENABLED(CONFIG_BT_CTLR_ADV_SYNC_PDU_LINK) ||
+	    (op == BT_HCI_LE_EXT_ADV_OP_UNCHANGED_DATA)) {
+		err = ull_adv_sync_pdu_set_clear(lll_sync, pdu_prev, pdu,
+						 ULL_ADV_PDU_HDR_FIELD_AD_DATA,
+						 0U, hdr_data);
+#if defined(CONFIG_BT_CTLR_ADV_SYNC_PDU_LINK)
+		/* No AD data overflow */
+		ad_len_overflow = 0U;
+#endif /* CONFIG_BT_CTLR_ADV_SYNC_PDU_LINK */
+	} else if (!IS_ENABLED(CONFIG_BT_CTLR_ADV_SYNC_PDU_LINK) ||
+		   (op == BT_HCI_LE_EXT_ADV_OP_FIRST_FRAG ||
+		    op == BT_HCI_LE_EXT_ADV_OP_COMPLETE_DATA)) {
+		/* Add AD Data and remove any prior presence of Aux Ptr */
+		err = ull_adv_sync_pdu_set_clear(lll_sync, pdu_prev, pdu,
+						 ULL_ADV_PDU_HDR_FIELD_AD_DATA,
+						 ULL_ADV_PDU_HDR_FIELD_AUX_PTR,
+						 hdr_data);
+#if defined(CONFIG_BT_CTLR_ADV_SYNC_PDU_LINK)
+		if (!err) {
+			/* Fragment into chain PDU if len > 191 bytes */
+			if (len > (PDU_AC_PAYLOAD_SIZE_MAX -
+				   PDU_AC_EXT_PAYLOAD_OVERHEAD)) {
```

Comment:
This way to calculate max PDU size for a single extended advertising PDU is used at leas in two places `PDU_AC_PAYLOAD_SIZE_MAX - PDU_AC_EXT_PAYLOAD_OVERHEAD1`. Maybe use of a single named macro that covers the subtraction? I would improve readability.